### ISSUE TITLE
Button announces multiple times when status changes to isPending

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#5519]
### Provide a general summary of the issue here

Following on #5245, Button announces more than once when the status changes to `isPending={true}` using Firefox with NVDA or JAWS.

### 🤔 Expected Behavior?

Button should not announce more than once when the status changes to `isPending={true}`.

### 😯 Current Behavior

Button announces more than once when the status changes to `isPending={true}` with NVDA and JAWS.

### 🖥️ Steps to Reproduce

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/88c4d560d552d6d911a71396759f9088b1df2199/storybook/index.html?path=/story/button--pending-spinner&providerSwitcher-express=false example in Firefox on Windows using JAWS or NVDA. 
2. Press any of the isPending button examples. 
3. When the button enters the pending state, the screen reader announces the button name and "pending" more than once.

### Version

https://github.com/adobe/react-spectrum/commit/9d8c74a344c9b7539f00fa58d0a02b062b6e830e

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What operating system are you using?

Windows, macOS

### 🧢 Your Company/Team

Adobe/Accessibility
